### PR TITLE
Fix#62

### PR DESCRIPTION
**Pull request by JeanMarieMineau**
Put the logic in the main loop.
The input reader and the whole display code are still in vblank are need to be reduced to the minimum.

### CODE REVIEW

File: doc/main.doc.md [hunk 1]
```diff
@@ -64,27 +64,53 @@ Each module `xxx` has a `xxx.init.s` file with its initialization code.
...
+	and a
+  jp nz,loop			; wait until lock = 0
+; \\\\ WAIT FOR VBLANK ////
+
+.INCLUDE "body.s"
+
+; //// ALLOW WBLANK TO UPDATE THE SCREEN \\\\
+	ld a,1
+	ld (lock),a    ; lock = 1
+; \\\\ ALLOW WBLANK TO UPDATE THE SCREEN ////
```
Comment (g33kex) [inline doc/main.doc.md:85]: What is WBlank?

File: doc/main.doc.md [hunk 2]
```diff
@@ -64,27 +64,53 @@ Each module `xxx` has a `xxx.init.s` file with its initialization code.
...
+The main logic in included in `body.s`.
+
+We use a look to synchronize the update of the values with the update of the screen. After each iteration we wait for a VBlank, and if we didn't finished the iteration, the VBlank interruption do nothing.
 ~~~
 ; ///////// MAIN LOOP \\\\\\\\\
 loop:
-	jr loop
+; //// WAIT FOR VBLANK \\\\
+	halt
+	ld a,(lock)
```
Comment (g33kex) [inline doc/main.doc.md:75]: `lock` is a bit vague for a global variable name

File: doc/main.doc.md [hunk 3]
```diff
@@ -64,27 +64,53 @@ Each module `xxx` has a `xxx.init.s` file with its initialization code.
...
-The VBLank function. **We include the `display` module first.**. The VLblank is a save time to edit the screen, and after this time, we can't edit the screen anymore, but we can still compute the values for the next frame.
-
-If the module `xxx` (`display` aside) needs to execute code at each iteration, the file `xxx.s` is included inside the file `body.s`.
+The VBlank function. This function is called with VBlank interrupt, but we only update the screen after each iteration of the loop. So, when the lock is not in the right state, we return at once. Else we run the code include in `xxx.vbl.s` files (the display module first, since we can only update the screen during the vblank time). The code included here has to be really fast. Long computations should be done in the loop.
 
 ~~~
 ; ///////// VBlank Interuption \\\\\\\\\
 VBlank:
 	push af
 	push hl
```
Comment (g33kex) [inline doc/main.doc.md:99]: In an interrupt, all registers are callee saved !  
You don't save bc nor de, and they might be modified in the VBlank interrupt.

File: doc/main.doc.md [hunk 4]
```diff
@@ -64,27 +64,53 @@ Each module `xxx` has a `xxx.init.s` file with its initialization code.
 
 ## Main Loop
 
-Do nothing, we work in `VBlank`.
+The main logic in included in `body.s`.
+
+We use a look to synchronize the update of the values with the update of the screen. After each iteration we wait for a VBlank, and if we didn't finished the iteration, the VBlank interruption do nothing.
```
Comment (g33kex) [inline doc/main.doc.md:69]: lock, not look.

File: src/main.s [hunk 5]
```diff
@@ -72,17 +77,43 @@ waitvlb: 					; wait for the line 144 to be refreshed:
...
+; \\\\\\\\\ MAIN LOOP /////////
+	jp loop
+; \\\\\\\\\ MAIN LOOP /////////
 
 
 ; ///////// VBlank Interuption \\\\\\\\\
+
 VBlank:
 	push af
 	push hl
```
Comment (g33kex) [inline src/main.s:102]: please push bc and de!!!!!

File: doc/main.doc.md (hunk 1 above)
Comment (JeanMarieMineau) [reply doc/main.doc.md:85]: It's short for VVVVVVVVVBlank

File: doc/main.doc.md (hunk 2 above)
Comment (JeanMarieMineau) [reply doc/main.doc.md:75]: Actually it's not global, it's local to the main file. (no global_ before) but yes, does VBlqnk_lock sound good?

File: doc/main.doc.md (hunk 3 above)
Comment (JeanMarieMineau) [reply doc/main.doc.md:99]: They are not supposed to be modified except between the loop iterations, but yes, better be safe than sorry.

File: doc/main.doc.md (hunk 4 above)
Comment (JeanMarieMineau) [reply doc/main.doc.md:69]: damn I did it again

File: src/main.s (hunk 5 above)
Comment (JeanMarieMineau) [reply src/main.s:102]: It will be done my lord

File: src/main.s [hunk 6]
```diff
@@ -115,6 +117,8 @@ VBlank:
 ; \\\\ REALLOW THE LOOP ////
 endVBlank:
 	pop hl
+	pop bc
+	pop de
 	pop af
```
Comment (g33kex) [inline src/main.s:122]: pop order doesn't match push order.

Comment (JeanMarieMineau) [reply src/main.s:122]: I really am stuppid

**Pull request merged**